components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -138,6 +146,57 @@ class BrowserIcons(
                 Logger.error("Could not install browser-icons extension", throwable)
             })
     }
+
+    /**
+     * Loads an icon asynchronously using [BrowserIcons] and then displays it in the [ImageView].
+     * If the view is detached from the window before loading is completed, then loading is cancelled.
+     *
+     * @param view [ImageView] to load icon into.
+     * @param request Load icon for this given [IconRequest].
+     * @param placeholder [Drawable] to display while icon is loading.
+     * @param error [Drawable] to display if loading fails.
+     */
+    fun loadIntoView(
+        view: ImageView,
+        request: IconRequest,
+        placeholder: Drawable? = null,
+        error: Drawable? = null
+    ) = scope.async(Dispatchers.Main) {
```

Comment:
Why is this using `async` instead of `launch`? It looks like we do not actually return anything from here. Even for the tests returning an actual `Job` from `launch()` may make more sense than an `Deferred<Unit>`.